src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    private static final String OPERATION_SUPPLY = "supply";
+    private static final String OPERATION_BUY = "buy";
+    private static final String OPERATION_RESULT = "result";
+    private static final String COMA_SEPARATOR = ",";
+    private static final String TEXT_OF_EXCEPTION = "File not found, check that the file path"
+            + " is correct and that the file still exists in the specified location";
+    private static final int AMOUNT_INDEX_0 = 0;
+    private static final int AMOUNT_INDEX_1 = 1;
+    private static final String DELIMITER = "\\W";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> dataFromBuffer = readFromFile(fromFileName);
+        int supplyAmount = calculateStatisticAmount(dataFromBuffer, OPERATION_SUPPLY);
+        int buyAmount = calculateStatisticAmount(dataFromBuffer, OPERATION_BUY);
+        String reportMessage = createReportMessage(supplyAmount, buyAmount);
+        writeStringToFile(toFileName, reportMessage);
+    }
+
+    private void writeStringToFile(String toFileName, String reportMessage) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName))) {
+            bufferedWriter.write(reportMessage);
+        } catch (IOException e) {
+            throw new RuntimeException(TEXT_OF_EXCEPTION, e);
+        }
+    }
+
+    private List<String> readFromFile(String fromFileName) {
+        List<String> dataFromBuffer = new ArrayList<>();
```

Comment:
Also let's avoid using collections in this task. We will learn and use them later :)